modules/oauth/src/main/java/org/mule/extension/oauth2/internal/authorizationcode/AuthorizationRequestHandler.java

```diff
@@ -51,44 +54,59 @@
   public static final String OAUTH_STATE_ID_FLOW_VAR_NAME = "resourceOwnerId";
 
   private Logger logger = LoggerFactory.getLogger(AuthorizationRequestHandler.class);
+
+  /**
+   * Scope required by this application to execute. Scopes define permissions over resources.
+   */
+  @Parameter
+  @Optional
   private String scopes;
-  private String state;
+
+  /**
+   * State parameter for holding state between the authentication request and the callback done by the oauth authorization server
+   * to the redirectUrl.
+   */
+  @Parameter
+  @Optional(defaultValue = "#[null]")
+  private Function<Event, String> state;
+
+  /**
+   * Identifier under which the oauth authentication attributes are stored (accessToken, refreshToken, etc).
+   * <p>
+   * This attribute is only required when the applications needs to access resources from more than one user in the OAuth
+   * authentication server.
+   */
+  @Parameter
+  @Optional(defaultValue = "#[null]")
```

Comment:
removing the default value leaves the Function as null, rather than a function that returns null.